ulib/FStar.Bytes.fsti

```diff
@@ -101,8 +101,22 @@ val set_byte:
 
 unfold let op_String_Assignment = set_byte
 
+val reveal_set_byte:
+    b:bytes
+  -> pos:u32 {U32.v pos < length b}
+  -> x:byte
+  -> Lemma
+    (reveal (set_byte b pos x) == Seq.upd (reveal b) (U32.v pos) x)
+    [SMTPat (reveal (set_byte b pos x))]
```

Comment:
Should the pattern be `set_byte b pos x`?

Reply:
I'm afraid such a pattern could trigger too often.

Reply:
I think removing the `reveal` is the right design ... otherwise you'll have to insert reveals yourself, just to make the pattern trigger (dealing with ghost effects in your code, etc.)

This is also the style elsewhere in this interface, e.g,. see `length_reveal`, one of whose patterns is just `len x` rather than `reveal (len x)`